### PR TITLE
Bulk Purchase and Package SDK

### PR DESCRIPTION
**Pull request by undersky0**

### CODE REVIEW

**Comment by lukeferguson** [comment]
Looks good @undersky0 . Just a few comments:
- Can you rename bbBulkPurchase to bbBulkPurchases to be consistent with the naming convention used for other directives
- Can you rename bbPackageItem to bbPackagesto be consistent with the naming convention used for other directives
- Can you remove unused dependencies from the controllers, for example 'halClient'
- Can you please (ngdoc) documentation for the new directives and services (see bbServices for an example)

**Comment by undersky0** [comment]
Thanks for checking, made the changes and added the ng-docs

File: src/core/javascripts/controllers/bulk_purchase.js.coffee [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  , (err) ->
+    $scope.setLoadedAndShowError($scope, err, 'Sorry, something went wrong')
+
+  $scope.init = (company) ->
+    $scope.booking_item ||= $scope.bb.current_item
+    ppromise = BulkPurchaseService.query($scope.bb.company)
+    ppromise.then (bulk_purchase) ->
+      $scope.bulk_purchases = []
+      $scope.bulk_purchases = bulk_purchase
+
```
Comment (lukeferguson) [inline src/core/javascripts/controllers/bulk_purchase.js.coffee]: double assignment!

File: src/core/javascripts/controllers/package_item.js.coffee [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+      $scope.init($scope.bb.company)
+  , (err) ->
+    $scope.setLoadedAndShowError($scope, err, 'Sorry, something went wrong')
+
+  $scope.init = (company) ->
+    $scope.booking_item ||= $scope.bb.current_item
+    promise = PackageItemService.query($scope.bb.company)
+    promise.then (package_item) ->
+      $scope.packages = []
+      $scope.packages = package_item
```
Comment (lukeferguson) [inline src/core/javascripts/controllers/package_item.js.coffee]: double assignment!

**Comment by lukeferguson** [comment]
Cool. Just a couple more issues - unnecessary double assignment of bulk_purchases and packages

**Pull request merged**